### PR TITLE
Remove connection as return value from Delete Connection

### PR DESCRIPTION
**Pull request by pliao28**
DeleteConnection method shouldn't return Connection. Removed Connection as return value and now returns Task.

### CODE REVIEW

File: test/WorkOSTests/Services/SSO/SSOServiceTest.cs [hunk 1]
```diff
@@ -321,5 +321,21 @@ public async void TestGetConnection()
...
         }
+
+        [Fact]
+        public async void TestDeleteConnection()
+        {
+            this.httpMock.MockResponse(
+                HttpMethod.Delete,
+                $"/connections/{this.mockConnection.Id}",
+                HttpStatusCode.OK,
+                RequestUtilities.ToJsonString(this.mockConnection));
```
Comment (maxdeviant) [inline test/WorkOSTests/Services/SSO/SSOServiceTest.cs:332]: Is this test correct?

It seems like the fixture is returning the connection is returning the connection in the response, when really it returns a 204 No Content with an empty body.

File: test/WorkOSTests/Services/SSO/SSOServiceTest.cs [hunk 2]
```diff
@@ -321,5 +321,21 @@ public async void TestGetConnection()
...
                 JsonConvert.SerializeObject(response));
         }
+
+        [Fact]
+        public async void TestDeleteConnection()
+        {
+            this.httpMock.MockResponse(
+                HttpMethod.Delete,
+                $"/connections/{this.mockConnection.Id}",
+                HttpStatusCode.OK,
```
Comment (maxdeviant) [inline test/WorkOSTests/Services/SSO/SSOServiceTest.cs:331]: This should be a 204 No Content to match the API:

```suggestion
                HttpStatusCode.NoContent,
```

Comment (pliao28) [reply test/WorkOSTests/Services/SSO/SSOServiceTest.cs:331]: HTTP 202 can also be returned by either operation and would imply that the instruction was accepted by the server, but not fully applied yet.

I can change the other tests to 204 instead if we prefer to do it that way.

Comment (pliao28) [reply test/WorkOSTests/Services/SSO/SSOServiceTest.cs]: Changed to 204 [9d43662](https://github.com/workos/workos-dotnet/pull/110/commits/9d43662a80d7fb96f931415c16e2c9d0c497f630), can change the other tests to also return 204 no content.

**Review by maxdeviant — APPROVED** [review]

**Pull request merged**